tests/conftest.py

```diff
@@ -37,7 +37,7 @@ def local_pulumi_login():
     run([pulumi_path, "logout"], check=False)
 
 
-@fixture(autouse=True)
+@fixture()
```

Comment:
I think we want autouse here, otherwise tests will log to your 'real' log file.

Reply:
This is the thing that causes tests to fail! Every test opens a new log file and doesn't close it until pytest is finished. This means that once you get to N tests, pytest fails due to too many open filehandles.

Perhaps we want a single common log file for tests? I can look into that.

Reply:
Hopefully f17fbec works

Reply:
No 🙃